anchor/src/client/base.ts

```diff
@@ -0,0 +1,360 @@
+    signer,
+    computeUnitLimit,
+    getPriorityFeeMicroLamports,
+    latestBlockhash,
+  }: {
+    tx: Transaction;
+    lookupTables?: Array<AddressLookupTableAccount> | [];
+    signer?: PublicKey;
+    computeUnitLimit?: number;
+    getPriorityFeeMicroLamports?: (tx: VersionedTransaction) => Promise<number>;
+    latestBlockhash?: BlockhashWithExpiryBlockHeight;
+  }): Promise<VersionedTransaction> {
+    lookupTables = lookupTables || [];
+    signer = signer || this.getSigner();
+
+    const instructions = tx.instructions;
+
+    if (!this.isPhantom()) {
+      // Set compute unit limit or autodetect by simulating the tx
+      if (!computeUnitLimit) {
+        try {
+          computeUnitLimit = await this.calcSimulationComputeUnits(
+            this.provider.connection,
+            instructions,
+            signer,
+            lookupTables,
+          );
+        } catch (e) {
+          //TODO
+          console.log(e);
+        }
+      }
+      if (computeUnitLimit) {
+        // ComputeBudgetProgram.setComputeUnitLimit costs 150 CUs
+        // Add 20%/50% more CUs to account for logs (mainnet logs are less verbose)
+        computeUnitLimit += 150;
+        if (this.isMainnet()) {
+          computeUnitLimit *= 1.2;
+        } else {
+          computeUnitLimit *= 1.5;
+        }
+        instructions.unshift(
+          ComputeBudgetProgram.setComputeUnitLimit({ units: computeUnitLimit }),
+        );
+      }
+    }
+
+    const recentBlockhash = (
+      latestBlockhash ? latestBlockhash : await this.blockhashWithCache.get()
+    ).blockhash;
+
+    let priorityFee = DEFAULT_PRIORITY_FEE;
+    if (getPriorityFeeMicroLamports) {
+      try {
+        const fee = await getPriorityFeeMicroLamports(
+          new VersionedTransaction(
+            new TransactionMessage({
+              payerKey: signer,
+              recentBlockhash,
+              instructions,
+            }).compileToV0Message(lookupTables),
+          ),
+        );
+        priorityFee = Math.ceil(fee);
+      } catch (e) {
+        console.log(e);
+      }
+    }
+
+    // Add the unit price instruction and return the final versioned transaction
+    instructions.unshift(
+      ComputeBudgetProgram.setComputeUnitPrice({
+        microLamports: priorityFee,
+      }),
+    );
+    return new VersionedTransaction(
+      new TransactionMessage({
+        payerKey: signer,
+        recentBlockhash,
+        instructions,
+      }).compileToV0Message(lookupTables),
+    );
+  }
+
+  isMainnet(): boolean {
+    return this.cluster === ClusterNetwork.Mainnet;
+  }
+
+  getSigner(): PublicKey {
+    const publicKey = this.wallet.publicKey;
+    if (!publicKey) {
+      throw new Error("Signer public key cannot be find.");
+    }
+    return publicKey;
+  }
+
+  isPhantom(): boolean {
+    if (!isBrowser) return false;
+    return (
+      // @ts-ignore
+      window?.phantom?.solana?.isPhantom && window?.phantom?.solana?.isConnected
+    );
```

Comment:
nit
Maybe add types to the top:
```ts
interface PhantomWindow extends Window {
  phantom?: {
    solana?: {
      isPhantom?: boolean;
      isConnected?: boolean;
    };
  };
}
```
Then you can do this:

```suggestion
  const phantomWindow = window as PhantomWindow;
  return !!(
    phantomWindow.phantom?.solana?.isPhantom && 
    phantomWindow.phantom?.solana?.isConnected
  );
```